xzes40/transformer/src/document.cpp

```diff
@@ -57,14 +52,21 @@ XALAN_USING_XERCES( XMLPlatformUtils  );
 // ----------------------------------------------------------------------------
 xzes::Document::Document( xzes::uri_t file_path )
 {
+    Cache::Cache storeList;
+
     // Set the file path to uri
     set_uri( file_path );
 
     // Set the Document ID
     set_id( );
 
     // Compile the document now that you know the type
-    compile( );
+    if(!storeList.search(uid)){
+        compile();
+        storeList.set( uid , doc , uri );
+    } else {
+        set_content( storeList.get( uid ));
+    }
```

Comment:
This is the major logical for using cache.
-If there is no file in the cache
    -compile() --means parse the file.
    -storeList.set -- means set the content to the cahce.
-If there is file in the cache
    -retrieve content from the cache, and then set it to `Document` class